test/test_functionalization.py

```diff
@@ -27,25 +27,29 @@ def are_aliased(x, y):
 # if/when functorch moves into core.
 # This is basically a crappy version of `functionalize()` for single-tensor-arg inputs.
 def _functionalize(f, *, reapply_views: bool, crossref: bool):
-    def wrapped(a):
+    def wrapped(*inputs):
         ctx = nullcontext()
         if crossref:
             ctx = enable_crossref_functionalize()
         with ctx:
-            input_functional = torch._to_functional_tensor(a)
-            input_functional.requires_grad = a.requires_grad
+            inputs_functional = []
+            for inpt in inputs:
+                input_functional = torch._to_functional_tensor(inpt)
+                input_functional.requires_grad = inpt.requires_grad
+                inputs_functional.append(input_functional)
             torch._enable_functionalization(reapply_views=reapply_views)
             try:
-                out = f(input_functional)
+                out = f(*inputs_functional)
             finally:
                 torch._disable_functionalization()
-            torch._sync(input_functional)
-            inpt_new = torch._from_functional_tensor(input_functional)
-            if inpt_new is not a:
-                # Existing deficiency in functionalize():
-                # we don't correctly mutate input metadata (yet?)
-                if inpt_new.shape == a.shape:
-                    a.copy_(inpt_new)
+            for inpt, input_functional in zip(inputs, inputs_functional):
```

Comment:
This can be beefed up a bit to support more containers with pytrees too. you can use:

```
inpts_flattened, _ = pytree.tree_flatten(inputs)
inputs_functional, _ = pyree.tree_flatten(inputs_functional)
```

before zipping them up